document use case for using load in client-side only contexts (#9864)
sveltejs/kit: documentation/docs/20-core-concepts/20-load.md

Thanks to the generated `$types` module, we get full type safety.

> A `load` function in a `+page.js` file runs both on the server and in the browser. If your `load` function should _always_ run on the server (because it uses private environment variables, for example, or accesses a database) then it would go in a `+page.server.js` instead. Conversely, if your `load` function should always run on the client-side (ex: you're building a [single page app](single-page-apps)) then you can add `export const ssr = false` to your `+page.js` or `+layout.js`, this will allow your `load` function to run in the client only.

Reviewer: Thank you, though I think this doesn't quite tell the full story — the `ssr` option defines rendering behaviour, and only indirectly affects where `load` runs. What I mean by this is that it's perfectly possible to have `export const ssr = false` in a `+page.server.js`, and your client-rendered app will `fetch` data from the server `load` function on startup.

This part of the docs probably isn't the best place for a lengthy explanation of all this stuff, but maybe something like this suffices?

```suggestion
A `load` function in a `+page.js` file runs both on the server and in the browser (unless combined with `export const ssr = false`, in which case it will [only run in the browser](https://kit.svelte.dev/docs/page-options#ssr)). If your `load` function should _always_ run on the server (because it uses private environment variables, for example, or accesses a database) then it would go in a `+page.server.js` instead.
```

Author: That looks good to me! I think a lengthy explanation could be placed in the SPA docs (https://kit.svelte.dev/docs/single-page-apps) to document a use case where you'd want to treat the `load` function as middleware before a route gets rendered in a SPA?

Ex: you want to check if your user is authenticated on the client side

Reviewer: Yeah, the SPA docs could probably make it clearer that if you have a non-prerendered route using `+page.server.js` or `+layout.server.js` then you're gonna need a server
